### PR TITLE
Integrate YAML LS

### PR DESCRIPTION
**Pull request by yanxi0830**
#56 
This integrates https://github.com/redhat-developer/yaml-language-server, and adds a preference page to associate schemas.

### CODE REVIEW

**Comment by mickaelistria** [comment]
Why do you think the preference page for schemas is necessary here? It seemed to me that file/schema mapping was implemented directly in the Language Server (so we wouldn't need extra support for it), isn't it @JPinkney ?

File: org.eclipse.wildwebdeveloper/language-servers/package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "vscode-json-languageserver": "./vscode-json-languageserver-1.0.1.tgz",
     "typescript": "2.8.3",
     "typescript-language-server": "0.3.7",
-    "node-debug2": "./node-debug2-1.25.5.tgz"
+    "node-debug2": "./node-debug2-1.25.5.tgz",
+    "yaml-language-server": "0.2.1"
```
Comment (mickaelistria) [inline org.eclipse.wildwebdeveloper/language-servers/package.json]: Why not 0.3.2 ?

**Comment by JPinkney** [comment]
@mickaelistria By default a lot of common schemas for YAML files are pulled in from the JSON schema store but if a user wants to set their own schema thats not in the schema store they'd need the preference page

**Comment by jabby** [comment]
@JPinkney I agree with you. The problem will be the same for the json language server. I try to provide a patch to vscode in order to use schemastore catalog and I realized that the  json language server provide a way to add user schema associations. 

Maybe a common preferences for user configuration will be a good idea.

**Comment by yanxi0830** [comment]
the vscode extension allows schema association through its common user/workspace settings, so I think the preference page would be the eclipse equivalent for doing so, unless there is a way to configure it globally like vscode without the preference page?

**Comment by mickaelistria** [comment]
> Maybe a common preferences for user configuration will be a good idea.

At least in Eclipse IDE, if a common preference page for JSon and Yaml schema can work, it can be a good thing to consider.
And if it backs to the same settings of underlying language servers, it'd be nice too.

> I think the preference page would be the eclipse equivalent for doing so, unless there is a way to configure it globally like vscode without the preference page?

Let's stick to a preference page at the moment, but my impression based on how XML schema were deployed is that users should basically never have to deal with a local schema association, since a good schema format is "self-described" (ie for a given file, there can be a flag describing what's the schema to use and where to retrieve it and that's a superior workflow than putting such responsibility in end-users' hands)

File: org.eclipse.wildwebdeveloper/language-servers/package.json (hunk 1 above)
Comment (mickaelistria) [reply org.eclipse.wildwebdeveloper/language-servers/package.json]: @yanxi0830 Your answer here is important as we cannot start the IP approval for yaml-ls before we're sure of which version to take (IP approvals are version specific).

Comment (yanxi0830) [reply org.eclipse.wildwebdeveloper/language-servers/package.json]: I was following the version used in https://github.com/redhat-developer/vscode-yaml, which was just updated from 0.2.1 to 0.3.2 a couple of days ago, so 0.3.2 would be the version to take.

**Comment by mickaelistria** [comment]
Here is the request to IP team for inclusion of Yaml LS.

**Comment by mickaelistria** [comment]
@yanxi0830 Would you mind squashing those commits as a single one to clarify history?

**Comment by yanxi0830** [comment]
> > Maybe a common preferences for user configuration will be a good idea.
> 
> At least in Eclipse IDE, if a common preference page for JSon and Yaml schema can work, it can be a good thing to consider.
> And if it backs to the same settings of underlying language servers, it'd be nice too.
> 

I can see that the Json LS and Yaml LS allows for similar settings of schema which can use the same schema-globPattern preference page. 

> there can be a flag describing what's the schema to use and where to retrieve it and that's a superior workflow than putting such responsibility in end-users' hands)

I think that at least with custom yaml files, the yaml ls requires users to sets schema. With this patch, I added the kubernetes schema by default since it breaks the diagnostics test if no schema are set.

**Pull request merged**